source/retryable-writes/tests/unified/notWritablePrimaryException.yml

```diff
@@ -0,0 +1,70 @@
+description: "NotWritablePrimary retryable writes error"
+
+schemaVersion: "1.3"
+
+runOnRequirements:
+  - minServerVersion: "4.2"
+    topologies: [replicaset, sharded, load-balanced]
+    auth: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      observeEvents: [commandStartedEvent, connectionCheckOutStartedEvent]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &databaseName retryable-notwritableprimary-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collectionName coll
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+
+tests:
+  - description: "should retry on insertOne"
+    operations:
+      - name: failPoint # fail the next connection establishment
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [insertOne]
+              errmsg: "not master"
+              errorCode: 1
```

Comment:
I don't recall seeing this in a previous iteration of the PR.

Are drivers still in the business of checking error strings for "not master"? `errorCode: 1` looks odd.

I suppose this has to do with [Use error messages to detect "not master" and "node is recovering"](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#id114)](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#use-error-messages-to-detect-not-master-and-node-is-recovering), but that says "When error codes are not available" so isn't there an error code we could/should use here instead of relying on string detection? As-is, I might wonder if a driver would first inspect `errorCode` and then design not to bother checking `errmsg`.

Also, I'll note that `errmsg` is not documented in either [failCommand parameters](https://github.com/mongodb/mongo/wiki/The-%22failCommand%22-fail-point#failcommand-parameters) or the equivalent docs in the [Unified Test Format spec](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#failcommand). Is that actually a supported `data` option? If so, should we update both the wiki and unified spec to discuss this?

Reply:
I pushed the new tests across two commits, the first one was a mistake, so the error code is now 10107 if you look at the latest. I agree we should drop the errmsg from the test b/c indeed the code should take precedence, I've done so. 
After a quick test, it looks like it is not a supported param in that it is ignored.